core/java/src/org/openda/exchange/dataobjects/NetcdfFileConcatenater.java

```diff
@@ -55,111 +75,261 @@ public static void main(String[] arguments) {
+			for (int i = 0; i < timesToBeAdded.length; i++) {
+				convertedTimesToBeAdded[i] = timesToBeAdded[i] + timeDiffInUnit;
+			}
+			if (convertedTimesToBeAdded[0] < timesTarget[timesTarget.length - 1]) throw new RuntimeException("File to be added has first time before last time of target file");
+			int totalTimesCombined = timeVariableTargetDimension.getLength() + timesToBeAdded.length;
+			if (convertedTimesToBeAdded[0] == timesTarget[timesTarget.length - 1]) {
+				addConcatenatedValueArraysToMaps(variableArraysMap, timeVariableArraysMap, targetVariable, timeVariableTarget, concatenateTimeVariable, targetLocationDimensionLength, targetValues, addedValues, timesTarget, convertedTimesToBeAdded, totalTimesCombined, true, useOldValueOnOverlap);
+			} else {
+				addConcatenatedValueArraysToMaps(variableArraysMap, timeVariableArraysMap, targetVariable, timeVariableTarget, concatenateTimeVariable, targetLocationDimensionLength, targetValues, addedValues, timesTarget, convertedTimesToBeAdded, totalTimesCombined, false, false);
+			}
+		}
+	}
+
+	private static double[][] getTargetValuesAsDoubles(Variable targetVariable, List<Dimension> targetDimensions) throws IOException {
+		Array read = targetVariable.read();
+		if (targetVariable.getDataType() == DataType.DOUBLE) return (double[][]) read.copyToNDJavaArray();
+		double[] doubleArray1D = (double[]) read.get1DJavaArray(Double.TYPE);
+		return split(doubleArray1D, targetDimensions.get(1).getLength());
+	}
+
+	private static void rewriteNetcdfFile(File targetNetcdfFile, NetcdfFile netcdfToAdd) throws IOException {
+		NetcdfFileWriter netcdfWriter = null;
+
+		try {
+			netcdfWriter = NetcdfFileWriter.createNew(NetcdfFileWriter.Version.netcdf3, targetNetcdfFile.getCanonicalPath());
+
+			redefineVariablesAndDimensions(netcdfToAdd, netcdfWriter);
+
+			netcdfWriter.create();
+
+			writeValues(netcdfToAdd, netcdfWriter);
+		} catch (InvalidRangeException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		} finally {
+			if (netcdfWriter != null) netcdfWriter.close();
+		}
+	}
+
+	private static void writeValues(NetcdfFile netcdfToAdd, NetcdfFileWriter netcdfWriter) throws IOException, InvalidRangeException {
+		List<Variable> variables = netcdfToAdd.getVariables();
+		for (Variable variable : variables) {
+			String fullNameEscaped = variable.getFullNameEscaped();
+			Variable netcdfFileVariable = netcdfWriter.findVariable(fullNameEscaped);
+			Array read = variable.read();
+			netcdfWriter.write(netcdfFileVariable, read);
+		}
+	}
+
+	private static void addConcatenatedValueArraysToMaps(Map<Variable, Array> variableArraysMap, Map<Variable, Array> timeVariableArraysMap, Variable targetVariable, Variable timeVariableTarget, boolean concatenateTimeVariable, int targetLocationDimensionLength, double[][] targetValues, double[][] addedValues, double[] timesTarget, double[] convertedTimesToBeAdded, int totalTimesCombined, boolean firstAddedTimeOverlapping, boolean useOldValueOnOverlap) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'addConcatenatedValueArraysToMaps(Map, Map, Variable, Variable, boolean, int, double, double, double, double, int, boolean, boolean)' has an NPath complexity of 1080](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=5271142)